src/utils.rs

```diff
@@ -40,11 +43,21 @@ pub(crate) fn extract_kate(extension: &HeaderExtension) -> (u16, u16, H256, Vec<
 	}
 }
 
-pub(crate) fn extract_app_lookup(extension: &HeaderExtension) -> &DataLookup {
-	match &extension {
+pub(crate) fn extract_app_lookup(extension: &HeaderExtension) -> DataLookup {
+	let compact = match &extension {
 		HeaderExtension::V1(v1::HeaderExtension { app_lookup, .. }) => app_lookup,
 		HeaderExtension::V2(v2::HeaderExtension { app_lookup, .. }) => app_lookup,
-	}
+	};
+
+	let size = compact.size;
+	let index = compact
+		.index
+		.iter()
+		.map(|item| DataLookupItem::new(AppId(item.app_id.0), item.start))
+		.collect::<Vec<_>>();
+
+	let compact = CompactDataLookup::new(size, index);
+	DataLookup::try_from(compact).expect("Invalid app lookup")
```

Comment:
We can return error instead of panic here, since this is used in `process_block` and `From` implementation for `BlockVerified` type, which both supports returning error.